src/sentry/incidents/models.py

```diff
@@ -263,6 +263,17 @@ def fetch_for_project(self, project):
         return self.filter(query_subscriptions__project=project)
 
 
+class AlertRuleEnvironment(Model):
+    __core__ = True
+
+    environment = FlexibleForeignKey("sentry.Environment", unique=True)
```

Comment:
I don't think you want this to be unique, you probably want a unique on `(environment, alert_rule)`